### PR TITLE
Enable banning in flow control

### PR DESCRIPTION
**Pull request by bartavelle**
This patch enables banning in flow control, and refactors limit checks a
bit so that common code is shared.

The currently noop test for banning flow control has also been
activated.

Signed-off-by: Simon Marechal <bartavelle@gmail.com>

### CODE REVIEW

**Comment by bartavelle** [comment]
The build is broken:

```
curiesync        | Traceback (most recent call last):
curiesync        |   File "/usr/local/bin/curieconfctl", line 3, in <module>
curiesync        |     from curieconf import cli
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/curieconf/cli/__init__.py", line 17, in <module>
curiesync        |     from curieconf import utils
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/curieconf/utils/__init__.py", line 7, in <module>
curiesync        |     from flask_restx import fields
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/flask_restx/__init__.py", line 4, in <module>
curiesync        |     from . import fields, reqparse, apidoc, inputs, cors
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/flask_restx/reqparse.py", line 19, in <module>
curiesync        |     from .model import Model
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/flask_restx/model.py", line 21, in <module>
curiesync        |     from jsonschema import Draft4Validator
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/jsonschema/__init__.py", line 21, in <module>
curiesync        |     from jsonschema._types import TypeChecker
curiesync        |   File "/usr/local/lib/python3.7/dist-packages/jsonschema/_types.py", line 6, in <module>
curiesync        |     from pyrsistent import pmap
curiesync        | ModuleNotFoundError: No module named 'pyrsistent'
```

**Comment by bartavelle** [comment]
The package is indeed installed, I do not know why it is not found :(

**Comment by bartavelle** [comment]
```
root@curiesync:/# ls /usr/lib/python3.7/site-packages/pyrsistent
__init__.py  __init__.pyi  __pycache__	_checked_types.py  _field_common.py  _helpers.py  _immutable.py  _pbag.py  _pclass.py  _pdeque.py  _plist.py  _pmap.py	_precord.py  _pset.py  _pvector.py  _toolz.py  _transformations.py  py.typed  typing.py  typing.pyi
```

**Review by Aviv-Galmidi — DISMISSED** [review]

File: curiefense/curieproxy/rust/curiefense/src/flow.rs [hunk 1]
```diff
@@ -85,29 +86,46 @@ pub fn flow_check(
...
-                            tags.insert(&elem.name);
-                            return Ok(SimpleDecision::Pass);
-                        }
-                        FlowResult::LastBlock => {
+                match build_redis_key(reqinfo, &elem.key, &elem.id, &elem.name) {
+                    Some(redis_key) => {
+                        // check for banned users : this will be triggered at every step of the checks
+                        let ban_key = get_ban_key(&redis_key);
+                        if is_banned(&mut cnx, &ban_key) {
+                            logs.debug("is banned!");
```
Comment (flaper87) [inline curiefense/curieproxy/rust/curiefense/src/flow.rs:98]: Is this debug line descriptive enough? Should it contain other things like the `ban_key`?

**Review by flaper87 — CHANGES_REQUESTED** [review]
needs rebase

File: curiefense/curieproxy/rust/curiefense/src/flow.rs (hunk 1 above)
Comment (bartavelle) [reply curiefense/curieproxy/rust/curiefense/src/flow.rs]: It should be better now!

**Review by Aviv-Galmidi — APPROVED** [review]

**Comment by bartavelle** [comment]
@flaper87 can you check if the change corresponds to what you wanted? :)

**Review by flaper87 — APPROVED** [review]
Although it needs rebase now :(

**Pull request closed without merging**